source/UpcommingMovies.Infra.TheMovieDb/Services/TheMovieDbServiceBase.cs

```diff
@@ -0,0 +1,123 @@
+        }
+
+
+        // Build the url including api_key, language and custom query parameters.
+        // api_key is taken from TheMovieDbConfig.ApiKey and language from CultureInfo.CurrentCulture.Name
+        private Uri BuildUri(string relativePath, IDictionary<string, string> query, bool supressLanguage)
+        {
+            if(query == null)
+            {
+                query = new Dictionary<string, string>();
+            }
+
+            query["api_key"] = _movieDbConfig.ApiKey;
+
+            if (!supressLanguage)
+            {
+                query["language"] = CultureInfo.CurrentCulture?.Name ?? "en-US";
+            }
+
+            string queryString = string.Join("&", 
+                    query.Select(q => string.Join("=", Uri.EscapeUriString(q.Key), Uri.EscapeUriString(q.Value))));
+
+            string relativePathWithStartSlashFree = relativePath.TrimStart('/');
+
+            return new Uri($"{_baseUriWithEndSlashFree}/{relativePathWithStartSlashFree}?{queryString}");
+        }
+
+        /// <summary>
+        /// Perform a get request for a specific relative path and query parameters. Is not necessary to supply api_key and language as query parameters.
+        /// </summary>
+        /// <param name="relativePath">The relative path to resource (without baseUri part).</param>
+        /// <param name="query">
+        /// A dictionary with parameter name as key and value to build the query string. 
+        /// api_key and language values are not necessary and should not supplied.
+        /// If supplied, they will be overridden by TheMovieDbConfig.ApiKey and CultureInfo.CurrentCulture.Name.
+        /// </param>
+        /// <param name="supressLanguage">If true it will not include language parameter on request final url querystring.</param>
+        /// <returns>Raw string response body if reponse status code is Ok (200 family)</returns>
+        /// <exception cref="ResourceUnauthorizedException">
+        /// Response gives a 401 (unauthorized) http status. This means that the api_key is not no longer valid or invalid.
+        /// </exception>
+        /// <exception cref="ResourceNotFoundException">
+        /// Response gives a 404 (not found) http status. Means that the request no longer exists, or the relativePath or baseUri is wrong.
+        /// </exception>
+        /// <exception cref="TooManyRequestsException">
+        /// Response gives a 429 (too many requests) http status. Means that the client did too much requests, 
+        /// take a look at themoviedb request rate limiting at https://developers.themoviedb.org/3/getting-started/request-rate-limiting 
+        /// </exception>
+        /// <exception cref="RemoteServiceException">
+        /// Response gives any http status but 200 family, 401, 404 or 429.
+        /// </exception>
+        protected async Task<string> GetAsync(string relativePath, IDictionary<string, string> query = null, bool supressLanguage = false)
```

Comment:
@arctouch-eltonsantana 
Connection exception handling is done at this point. The exceptions type thrown by this method extends CoreException which all will being cautch by the UI logic.